stis_specutils_loader.ipynb

```diff
@@ -0,0 +1,514 @@
+    "        search_for = [\n",
+    "            os.path.join(dirname, basename.rsplit('_x1c.fits',1)[0] + '_flt.fits'),\n",
+    "            os.path.join(dirname, basename.rsplit('_s1c.fits',1)[0] + '_flt.fits'),\n",
+    "            os.path.join(dirname, basename.rsplit('_x1d.fits',1)[0] + '_flt.fits'),\n",
+    "            os.path.join(dirname, basename.rsplit('_sx1.fits',1)[0] + '_flt.fits')]\n",
+    "        search_for.sort(key=len)\n",
+    "        search_for = search_for[0]\n",
+    "        flt_file = glob.glob(search_for)\n",
+    "        \n",
+    "        if len(flt_file) != 1:\n",
+    "            print ('WARNING:  Unique FLT file not found in X1D\\'s directory:  {}'.format(search_for))\n",
+    "            wcs = None\n",
+    "        else:\n",
+    "            flt_file = flt_file[0]\n",
+    "            # WCS must be generated from the FLT file:\n",
+    "            wcs = WCS(fits.getheader(flt_file, ext=('SCI',ext)))\n",
+    "        \n",
+    "        if weight is not None:\n",
+    "            # Require weight array to be 2D:\n",
+    "            weight = np.array(weight).squeeze()\n",
+    "            if len(np.shape(weight)) == 1:\n",
+    "                weight = np.array([weight])\n",
+    "            #print (np.shape(weight), np.shape(tab['WAVELENGTH']))\n",
+    "            if np.shape(tab['WAVELENGTH']) != np.shape(weight):\n",
+    "                raise Exception('Dimensionality of (optional) WEIGHT array must match '\n",
+    "                                'WAVELENGTH/FLUX/ERROR/DQ arrays.')\n",
+    "        \n",
+    "        spectra = []  # Will become a spectrumCollection later!\n",
+    "        for i in range(len(tab)):\n",
+    "            flags = tab[i]['DQ'] & sdqflags\n",
+    "            uncertainty = StdDevUncertainty(tab[i]['ERROR']) #* u(header[1].data.columns['ERROR'].unit)\n",
+    "            wave = tab[i]['WAVELENGTH'] * u(f[1].data.columns['WAVELENGTH'].unit.replace('Angstroms', 'Angstrom'))\n",
+    "            flux = tab[i]['FLUX'] * u(f[1].data.columns['FLUX'].unit)\n",
+    "            meta = {'header': header, 'sdqflags': sdqflags, 'flags': flags, 'dq': tab[i]['DQ'], \n",
+    "                    'sporder': tab[i]['SPORDER']}\n",
+    "            if weight is not None:\n",
+    "                meta['weight'] = weight[i,:]\n",
+    "            \n",
+    "            # WCS shouldn't overtake spectral_axis here!  Leave out WCS for now:\n",
+    "            spectra.append(Spectrum1D(spectral_axis=wave, flux=flux, wcs=None, uncertainty=uncertainty, meta=meta))\n",
+    "\n",
+    "    return spectra"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "WARNING:root:GWCS does not store rest frequency information. Please define the rest value explicitly in the `Spectrum1D` object.\n",
```

Comment:
If this notebook is going in the repo, it would be helpful to catch & silence these warnings.

Reply:
Oops, it's not supposed to be there. It was test notebook from the STIS team.